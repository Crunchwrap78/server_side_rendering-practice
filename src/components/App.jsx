import React, { Component } from 'react';
import Header from './Header';
import Sidebar from './Sidebar';
import ContentList from './ContentList';
import listingsData from '../data/listings.js'

class App extends Component {
  constructor(props){
  	super(props);
  	this.state = {
      city: 'All',
      rooms: 0,
      listingsData,
      homeType: 'All',
      min_price: 0,
      max_price: 10000000,
      min_floor_space: 0,
      max_floor_space: 4000,
      elevator: false,
      basement: false,
      swimming_pool: false,
      gym: false,
      filteredData: listingsData,
      populateDropDown: '',
      sortby: 'price-dsc',
      search: '',
      view: 'long'
    };

    this.handleChange = this.handleChange.bind(this);
    this.filterData = this.filterData.bind(this);
    this.populateOptions = this.populateOptions.bind(this);
    this.changeView = this.changeView.bind(this)
  }

  componentWillMount(){
    const listingsData = this.state.listingsData.sort((a, b) => {
      return a.price - b.price
    });
    this.setState({
      listingsData
    });
  }

  changeView(viewName){
    this.setState({
      view: viewName
    })
  }

  handleChange(event) {
    const name = event.target.name
    const value = event.target.type === 'checkbox' ? event.target.checked : event.target.value
    this.setState({
      [ name ]: value
    }, ()=> {
      this.filterData()
    })

  }

  filterData() {
    let newData = this.state.listingsData.filter(item =>
      item.price >= this.state.min_price && item.price <= this.state.max_price && item.floorSpace >= this.state.min_floor_space && item.floorSpace <= this.state.max_floor_space && item.rooms >= this.state.rooms
    );

    if(this.state.city !== 'All') {
      newData = newData.filter(item => item.city === this.state.city)
    }

    if(this.state.homeType !== 'All') {
      newData = newData.filter(item => item.homeType === this.state.homeType)
    }

    if(this.state.sortby == 'price-dsc') {
      newData.sort((a, b) => b.price - a.price);
    }

    if(this.state.sortby == 'price-asc') {
      newData.sort((a, b) => a.price - b.price);
    }

    if(this.state.search != ''){
      newData = newData.filter((item) => {
        const city = item.city.toLowerCase()
        const searchText = this.state.search.toLowerCase()
        const n = city.match(searchText)

        if(n != null) {
          return true
        }
      })
    }

    this.setState({filteredData: newData});
  }

  populateOptions() {
    let cities = this.state.listingsData.map(item => item.city );
    cities = new Set(cities);
    cities = [...cities];

    let homeTypes = this.state.listingsData.map(item => item.homeType);
    homeTypes = new Set(homeTypes);
    homeTypes = [...homeTypes];

    let numOfRooms = this.state.listingsData.map(item => item.rooms);
    numOfRooms = new Set(numOfRooms);
    numOfRooms = [...numOfRooms];
    numOfRooms.sort();

    this.setState({
      populateDropDown: {
        homeTypes,
        numOfRooms,
        cities
      }
    });
  }

  render() {
    return (
      <div>
        <Header />
        <section id="content-area">
          <Sidebar
            handleChange={this.handleChange}
            globalState={this.state}
            populateOption={this.populateOptions}
          />
          <ContentList
            handleChange={this.handleChange}
            listingsData={this.state.filteredData}
            globalState={this.state}
            changeView={this.changeView}
          />
        </section>
      </div>
    )
  }
}

export default App;
