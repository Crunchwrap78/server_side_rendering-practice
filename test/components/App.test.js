import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/components/App';
import listingsData from '../../src/data/listings.js';

let wrapper;

test('should render App correctly', () => {
  wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
