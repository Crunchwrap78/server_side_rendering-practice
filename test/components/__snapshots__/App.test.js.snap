// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<div>
  <Header />
  <section
    id="content-area"
  >
    <Sidebar
      globalState={
        Object {
          "basement": false,
          "city": "All",
          "elevator": false,
          "filteredData": Array [
            Object {
              "address": "3 universal st",
              "city": "Miami",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 1430,
              "homeType": "Condo",
              "image": "https://www.adinahotels.com/wp-content/uploads/sites/4/2016/07/adina-melbourne-flinders-street-apartment-hotel-one-and-two-bedroom-apartment-2-2013.jpg",
              "price": 24666,
              "rooms": 2,
              "state": "FL",
            },
            Object {
              "address": "889 beemore st",
              "city": "Newark",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 1000,
              "homeType": "Studio",
              "image": "http://media.equityapartments.com/images/q_50/f_auto/fl_lossy/685-28/oak-park-apartment-homes-kitchen",
              "price": 80000,
              "rooms": 0,
              "state": "NJ",
            },
            Object {
              "address": "730 gates ave",
              "city": "Springfield",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2000,
              "homeType": "Room",
              "image": "http://www.carolinacouture.com/wood/wp-content/uploads/2014/09/apartment-contemporary-apartment-in-taiwan-by-fertility-design-and-mirror-glass-accent-stylish-and-elegant-apartment-designs-tiny-ass-apartment-design-ideas-apartment-interior.jpg",
              "price": 150000,
              "rooms": 1,
              "state": "IN",
            },
            Object {
              "address": "20-34 grand ave",
              "city": "Ridgewood",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2000,
              "homeType": "Apartment",
              "image": "http://media.equityapartments.com/images/c_crop,x_0,y_0,w_1920,h_1080/c_fill,w_1920,h_1080/q_80/4206-28/the-kelvin-apartments-exterior.jpg",
              "price": 220000,
              "rooms": 3,
              "state": "NY",
            },
            Object {
              "address": "2 main st",
              "city": "Bedstuy",
              "extras": Array [
                "elevator",
                "pool",
              ],
              "floorSpace": 2000,
              "homeType": "Multi Home",
              "image": "http://www.theproserve.com/wp-content/uploads/2013/03/apartment-complex-insurance.jpg",
              "price": 220000,
              "rooms": 2,
              "state": "NY",
            },
            Object {
              "address": "1 president plaza",
              "city": "Richmond",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2400,
              "homeType": "Single Home",
              "image": "https://c.o0bg.com/rf/image_960w/Boston/2011-2020/2017/01/09/BostonGlobe.com/Lifestyle/Images/doherty_10names04_liv.jpg",
              "price": 345355,
              "rooms": 1,
              "state": "VA",
            },
            Object {
              "address": "43 hollywood blvd",
              "city": "Los Angeles",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 3000,
              "homeType": "Condo",
              "image": "http://cdn.freshome.com/wp-content/uploads/2016/02/design-modern-apartment-1-1025x450.jpg",
              "price": 603000,
              "rooms": 3,
              "state": "CA",
            },
          ],
          "gym": false,
          "homeType": "All",
          "listingsData": Array [
            Object {
              "address": "3 universal st",
              "city": "Miami",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 1430,
              "homeType": "Condo",
              "image": "https://www.adinahotels.com/wp-content/uploads/sites/4/2016/07/adina-melbourne-flinders-street-apartment-hotel-one-and-two-bedroom-apartment-2-2013.jpg",
              "price": 24666,
              "rooms": 2,
              "state": "FL",
            },
            Object {
              "address": "889 beemore st",
              "city": "Newark",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 1000,
              "homeType": "Studio",
              "image": "http://media.equityapartments.com/images/q_50/f_auto/fl_lossy/685-28/oak-park-apartment-homes-kitchen",
              "price": 80000,
              "rooms": 0,
              "state": "NJ",
            },
            Object {
              "address": "730 gates ave",
              "city": "Springfield",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2000,
              "homeType": "Room",
              "image": "http://www.carolinacouture.com/wood/wp-content/uploads/2014/09/apartment-contemporary-apartment-in-taiwan-by-fertility-design-and-mirror-glass-accent-stylish-and-elegant-apartment-designs-tiny-ass-apartment-design-ideas-apartment-interior.jpg",
              "price": 150000,
              "rooms": 1,
              "state": "IN",
            },
            Object {
              "address": "20-34 grand ave",
              "city": "Ridgewood",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2000,
              "homeType": "Apartment",
              "image": "http://media.equityapartments.com/images/c_crop,x_0,y_0,w_1920,h_1080/c_fill,w_1920,h_1080/q_80/4206-28/the-kelvin-apartments-exterior.jpg",
              "price": 220000,
              "rooms": 3,
              "state": "NY",
            },
            Object {
              "address": "2 main st",
              "city": "Bedstuy",
              "extras": Array [
                "elevator",
                "pool",
              ],
              "floorSpace": 2000,
              "homeType": "Multi Home",
              "image": "http://www.theproserve.com/wp-content/uploads/2013/03/apartment-complex-insurance.jpg",
              "price": 220000,
              "rooms": 2,
              "state": "NY",
            },
            Object {
              "address": "1 president plaza",
              "city": "Richmond",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2400,
              "homeType": "Single Home",
              "image": "https://c.o0bg.com/rf/image_960w/Boston/2011-2020/2017/01/09/BostonGlobe.com/Lifestyle/Images/doherty_10names04_liv.jpg",
              "price": 345355,
              "rooms": 1,
              "state": "VA",
            },
            Object {
              "address": "43 hollywood blvd",
              "city": "Los Angeles",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 3000,
              "homeType": "Condo",
              "image": "http://cdn.freshome.com/wp-content/uploads/2016/02/design-modern-apartment-1-1025x450.jpg",
              "price": 603000,
              "rooms": 3,
              "state": "CA",
            },
          ],
          "max_floor_space": 4000,
          "max_price": 10000000,
          "min_floor_space": 0,
          "min_price": 0,
          "populateDropDown": "",
          "rooms": 0,
          "search": "",
          "sortby": "price-dsc",
          "swimming_pool": false,
          "view": "long",
        }
      }
      handleChange={[Function]}
      populateOption={[Function]}
    />
    <ContentList
      changeView={[Function]}
      globalState={
        Object {
          "basement": false,
          "city": "All",
          "elevator": false,
          "filteredData": Array [
            Object {
              "address": "3 universal st",
              "city": "Miami",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 1430,
              "homeType": "Condo",
              "image": "https://www.adinahotels.com/wp-content/uploads/sites/4/2016/07/adina-melbourne-flinders-street-apartment-hotel-one-and-two-bedroom-apartment-2-2013.jpg",
              "price": 24666,
              "rooms": 2,
              "state": "FL",
            },
            Object {
              "address": "889 beemore st",
              "city": "Newark",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 1000,
              "homeType": "Studio",
              "image": "http://media.equityapartments.com/images/q_50/f_auto/fl_lossy/685-28/oak-park-apartment-homes-kitchen",
              "price": 80000,
              "rooms": 0,
              "state": "NJ",
            },
            Object {
              "address": "730 gates ave",
              "city": "Springfield",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2000,
              "homeType": "Room",
              "image": "http://www.carolinacouture.com/wood/wp-content/uploads/2014/09/apartment-contemporary-apartment-in-taiwan-by-fertility-design-and-mirror-glass-accent-stylish-and-elegant-apartment-designs-tiny-ass-apartment-design-ideas-apartment-interior.jpg",
              "price": 150000,
              "rooms": 1,
              "state": "IN",
            },
            Object {
              "address": "20-34 grand ave",
              "city": "Ridgewood",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2000,
              "homeType": "Apartment",
              "image": "http://media.equityapartments.com/images/c_crop,x_0,y_0,w_1920,h_1080/c_fill,w_1920,h_1080/q_80/4206-28/the-kelvin-apartments-exterior.jpg",
              "price": 220000,
              "rooms": 3,
              "state": "NY",
            },
            Object {
              "address": "2 main st",
              "city": "Bedstuy",
              "extras": Array [
                "elevator",
                "pool",
              ],
              "floorSpace": 2000,
              "homeType": "Multi Home",
              "image": "http://www.theproserve.com/wp-content/uploads/2013/03/apartment-complex-insurance.jpg",
              "price": 220000,
              "rooms": 2,
              "state": "NY",
            },
            Object {
              "address": "1 president plaza",
              "city": "Richmond",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2400,
              "homeType": "Single Home",
              "image": "https://c.o0bg.com/rf/image_960w/Boston/2011-2020/2017/01/09/BostonGlobe.com/Lifestyle/Images/doherty_10names04_liv.jpg",
              "price": 345355,
              "rooms": 1,
              "state": "VA",
            },
            Object {
              "address": "43 hollywood blvd",
              "city": "Los Angeles",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 3000,
              "homeType": "Condo",
              "image": "http://cdn.freshome.com/wp-content/uploads/2016/02/design-modern-apartment-1-1025x450.jpg",
              "price": 603000,
              "rooms": 3,
              "state": "CA",
            },
          ],
          "gym": false,
          "homeType": "All",
          "listingsData": Array [
            Object {
              "address": "3 universal st",
              "city": "Miami",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 1430,
              "homeType": "Condo",
              "image": "https://www.adinahotels.com/wp-content/uploads/sites/4/2016/07/adina-melbourne-flinders-street-apartment-hotel-one-and-two-bedroom-apartment-2-2013.jpg",
              "price": 24666,
              "rooms": 2,
              "state": "FL",
            },
            Object {
              "address": "889 beemore st",
              "city": "Newark",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 1000,
              "homeType": "Studio",
              "image": "http://media.equityapartments.com/images/q_50/f_auto/fl_lossy/685-28/oak-park-apartment-homes-kitchen",
              "price": 80000,
              "rooms": 0,
              "state": "NJ",
            },
            Object {
              "address": "730 gates ave",
              "city": "Springfield",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2000,
              "homeType": "Room",
              "image": "http://www.carolinacouture.com/wood/wp-content/uploads/2014/09/apartment-contemporary-apartment-in-taiwan-by-fertility-design-and-mirror-glass-accent-stylish-and-elegant-apartment-designs-tiny-ass-apartment-design-ideas-apartment-interior.jpg",
              "price": 150000,
              "rooms": 1,
              "state": "IN",
            },
            Object {
              "address": "20-34 grand ave",
              "city": "Ridgewood",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2000,
              "homeType": "Apartment",
              "image": "http://media.equityapartments.com/images/c_crop,x_0,y_0,w_1920,h_1080/c_fill,w_1920,h_1080/q_80/4206-28/the-kelvin-apartments-exterior.jpg",
              "price": 220000,
              "rooms": 3,
              "state": "NY",
            },
            Object {
              "address": "2 main st",
              "city": "Bedstuy",
              "extras": Array [
                "elevator",
                "pool",
              ],
              "floorSpace": 2000,
              "homeType": "Multi Home",
              "image": "http://www.theproserve.com/wp-content/uploads/2013/03/apartment-complex-insurance.jpg",
              "price": 220000,
              "rooms": 2,
              "state": "NY",
            },
            Object {
              "address": "1 president plaza",
              "city": "Richmond",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 2400,
              "homeType": "Single Home",
              "image": "https://c.o0bg.com/rf/image_960w/Boston/2011-2020/2017/01/09/BostonGlobe.com/Lifestyle/Images/doherty_10names04_liv.jpg",
              "price": 345355,
              "rooms": 1,
              "state": "VA",
            },
            Object {
              "address": "43 hollywood blvd",
              "city": "Los Angeles",
              "extras": Array [
                "elevator",
                "gym",
              ],
              "floorSpace": 3000,
              "homeType": "Condo",
              "image": "http://cdn.freshome.com/wp-content/uploads/2016/02/design-modern-apartment-1-1025x450.jpg",
              "price": 603000,
              "rooms": 3,
              "state": "CA",
            },
          ],
          "max_floor_space": 4000,
          "max_price": 10000000,
          "min_floor_space": 0,
          "min_price": 0,
          "populateDropDown": "",
          "rooms": 0,
          "search": "",
          "sortby": "price-dsc",
          "swimming_pool": false,
          "view": "long",
        }
      }
      handleChange={[Function]}
      listingsData={
        Array [
          Object {
            "address": "3 universal st",
            "city": "Miami",
            "extras": Array [
              "elevator",
              "gym",
            ],
            "floorSpace": 1430,
            "homeType": "Condo",
            "image": "https://www.adinahotels.com/wp-content/uploads/sites/4/2016/07/adina-melbourne-flinders-street-apartment-hotel-one-and-two-bedroom-apartment-2-2013.jpg",
            "price": 24666,
            "rooms": 2,
            "state": "FL",
          },
          Object {
            "address": "889 beemore st",
            "city": "Newark",
            "extras": Array [
              "elevator",
              "gym",
            ],
            "floorSpace": 1000,
            "homeType": "Studio",
            "image": "http://media.equityapartments.com/images/q_50/f_auto/fl_lossy/685-28/oak-park-apartment-homes-kitchen",
            "price": 80000,
            "rooms": 0,
            "state": "NJ",
          },
          Object {
            "address": "730 gates ave",
            "city": "Springfield",
            "extras": Array [
              "elevator",
              "gym",
            ],
            "floorSpace": 2000,
            "homeType": "Room",
            "image": "http://www.carolinacouture.com/wood/wp-content/uploads/2014/09/apartment-contemporary-apartment-in-taiwan-by-fertility-design-and-mirror-glass-accent-stylish-and-elegant-apartment-designs-tiny-ass-apartment-design-ideas-apartment-interior.jpg",
            "price": 150000,
            "rooms": 1,
            "state": "IN",
          },
          Object {
            "address": "20-34 grand ave",
            "city": "Ridgewood",
            "extras": Array [
              "elevator",
              "gym",
            ],
            "floorSpace": 2000,
            "homeType": "Apartment",
            "image": "http://media.equityapartments.com/images/c_crop,x_0,y_0,w_1920,h_1080/c_fill,w_1920,h_1080/q_80/4206-28/the-kelvin-apartments-exterior.jpg",
            "price": 220000,
            "rooms": 3,
            "state": "NY",
          },
          Object {
            "address": "2 main st",
            "city": "Bedstuy",
            "extras": Array [
              "elevator",
              "pool",
            ],
            "floorSpace": 2000,
            "homeType": "Multi Home",
            "image": "http://www.theproserve.com/wp-content/uploads/2013/03/apartment-complex-insurance.jpg",
            "price": 220000,
            "rooms": 2,
            "state": "NY",
          },
          Object {
            "address": "1 president plaza",
            "city": "Richmond",
            "extras": Array [
              "elevator",
              "gym",
            ],
            "floorSpace": 2400,
            "homeType": "Single Home",
            "image": "https://c.o0bg.com/rf/image_960w/Boston/2011-2020/2017/01/09/BostonGlobe.com/Lifestyle/Images/doherty_10names04_liv.jpg",
            "price": 345355,
            "rooms": 1,
            "state": "VA",
          },
          Object {
            "address": "43 hollywood blvd",
            "city": "Los Angeles",
            "extras": Array [
              "elevator",
              "gym",
            ],
            "floorSpace": 3000,
            "homeType": "Condo",
            "image": "http://cdn.freshome.com/wp-content/uploads/2016/02/design-modern-apartment-1-1025x450.jpg",
            "price": 603000,
            "rooms": 3,
            "state": "CA",
          },
        ]
      }
    />
  </section>
</div>
`;
